Automattic/wp-calypso
# Tweak pricing page i2 layout

### Changes proposed in this Pull Request

This PR does the following:
- it only shown bundle ribbons to plan cards
- it aligns the plan cards vertically (when a row contains 3 cards)
- it collapses/expands all plans cards altogether (when a row contains 3 cards)

1196341175636977-as-1198920526622510

### Implementation notes

- Created a new grid component for i2
- Extracted reused functions and config in `product-grid` folder

### Testing instructions

- Download the PR
- Run Calypso and cloud concurrently
- Make sure the variant `v2 - slide outs` is selected in the `jetpackConversionRateOptimization` A/B test
- Visit the plans page in both Calypso and cloud
- Check that only plan cards have a ribbon
- When the screen is wide enough to include 3 cards, check that the plan cards are expanded and aligned, that they collapse/expand together, and that a border separates plans from products (see below). Check that product cards behave normally.
- On desktop, when the screen isn't wide enough (1 or 2 cards), just check the border (see below)
- On mobile, check that there's no border (see below)
- Switch back to the v1 variant and check that the page looks like production

### Screenshots

#### Desktop - 3 cards
<img width="1078" alt="Screen Shot 2020-10-27 at 4 46 12 PM" src="https://user-images.githubusercontent.com/1620183/97359968-f2dff000-1873-11eb-9267-24e472e7019d.png">

_Plan cards are expanded by default. Feature toggles and CTA buttons are aligned. A border separates plans from products._

#### Desktop - 2 cards
<img width="817" alt="Screen Shot 2020-10-27 at 4 48 22 PM" src="https://user-images.githubusercontent.com/1620183/97360204-3f2b3000-1874-11eb-9dc0-b9fa683db1a2.png">

_Plan cards are collapsed. A border separates plans from products._

#### Desktop - 1 card
<img width="583" alt="Screen Shot 2020-10-27 at 4 49 18 PM" src="https://user-images.githubusercontent.com/1620183/97360305-63870c80-1874-11eb-90d2-4d59e3278c4a.png">

_Plan cards are collapsed. A border separates plans from products._

#### Mobile
<img width="392" alt="Screen Shot 2020-10-27 at 4 49 50 PM" src="https://user-images.githubusercontent.com/1620183/97360350-76014600-1874-11eb-8e14-6506f5115f2b.png">

_Plan cards are collapsed. No border separates plans from products._
